This series makes the build of the whole layer more reliable

fmoessbauer opened the pull request.

fmoessbauer committed:
ci: set code.siemens.com proxies

We explicitly set the CSC proxies for our internal CI builds.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
chore: update kas to version 3.1

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
fix: run bazel-fetch against local apt list

This fixes a bug that lead to an inconsitent apt state between the one
used by sbuild and the one used to prepare the package.

Reported-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
fix(dpkg-bazel): apply quilt patches before fetching

This patch reworks the bazel-fetch task by first applying the quilt
patches, then fetching and then removing the patches again.
By that, the sources archive never contains patched code which later
might break when the patches are applied again in the build task.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
chore: speedup compression of .orig files

This patch reduces the compression factor for the .orig.xz files
generated for bazel packages. These are only used to create the source
package and do not require a very good compression.

This speeds up the build significantly, especially on systems with
little memory and cores.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

jan-kiszka commented on .gitlab-ci.yml:
```diff
@@ -16,6 +16,7 @@ variables:
   SSTATE_LOCATION: "${RUNNER_SSTATE}/${CI_PROJECT_NAME}"
   SSTATE_MIRRORS: "file://.* ${SSTATE_LOCATION}/PATH;downloadfilename=PATH"
   SSTATE_MAX_AGE: "3d"
+  DISTRO_APT_PREMIRRORS: 'http://deb\.debian\.org https://debian.inf.tu-dresden.de\n'
```
Why? I know reasons, but they should go into the commit message.

fmoessbauer replied:
Added the reason.

jan-kiszka commented on .gitlab-ci.yml:
```diff
@@ -22,7 +22,7 @@ variables:
 .kas-build:
   stage: build
-  image: ghcr.io/siemens/kas/kas-isar:3.0.2
+  image: ghcr.io/siemens/kas/kas-isar:3.1
```
What about the kas-container copy in this layer?

fmoessbauer replied:
Good catch. Fixed.

jan-kiszka replied:
Then maybe also rename the subject as you are no longer only targeting "ci:".

jan-kiszka commented on .gitlab-ci.yml:
```diff
@@ -16,6 +16,7 @@ variables:
   SSTATE_LOCATION: "${RUNNER_SSTATE}/${CI_PROJECT_NAME}"
   SSTATE_MIRRORS: "file://.* ${SSTATE_LOCATION}/PATH;downloadfilename=PATH"
   SSTATE_MAX_AGE: "3d"
+  DISTRO_APT_PREMIRRORS: 'http://deb\.debian\.org https://debian.inf.tu-dresden.de\n'
```
If that only affects our CI setup, then better make this a CI variable that is using this value only in our local project settings.

jan-kiszka commented:
Looks good otherwise and works as well (on IOT2050).

fmoessbauer committed:
feat(bazel) include taskhash in download artifact

This patch adds the BB_TASKHASH to the hash of the cached source
package. By that, changes to the fetch task itself are also reflected.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>
